langs/python.go

```diff
@@ -75,8 +75,7 @@ func (h *PythonLangHelper) DockerfileBuildCmds() []string {
 	r = append(r, "ADD . /function/")
 	if exists("requirements.txt") {
```

Comment:
not 100% sure it's right but I think you can save running pip every time by pre-loading requirements.txt and running pip before copying the rest of the functions context 

```
r := []string{}
	if exists("requirements.txt") {
		r = append(r, "ADD requirements.txt /function/requirements.txt")
		r = append(r, `
RUN pip3 install --target /python/  --disable-pip-version-check  --no-cache --no-cache-dir -r requirements.txt &&\
    rm -fr ~/.cache/pip /tmp* requirements.txt func.yaml Dockerfile .venv`)
	}

	r = append(r, "ADD . /function/")

	return r
```

Reply:
on my mac at least this lets the pip stuff get cached in docker build.